app/models/post.rb

```diff
@@ -0,0 +1,13 @@
+class Post < ApplicationRecord
+  belongs_to :user, foreign_key: 'author_id', counter_cache: true
+  has_many :likes
+  has_many :comments
+
+  def update_counter
+    user.update(post_count: user.posts.size)
+  end
+
+  def recent_comment
+    comments.where(post_id: self[:id]).order(created_at: :desc).limit(5)
+  end
+end
```

Comment:
- [ ] Great job on this implementation. however It would be nice to use one of the Rails active callbacks `after_save` to call the `#update_counter` method every time a new post has been saved to database, then make the `update_counter` method private. You would have to rearrange your code as follows:

e.g

```
class Post < ApplicationRecord
  belongs_to :user, foreign_key: 'author_id', counter_cache: true
  has_many :likes
  has_many :comments

  after_save :update_counter


  def recent_comment
    comments.where(post_id: self[:id]).order(created_at: :desc).limit(5)
  end

  private

  def update_counter
    user.update(post_count: user.posts.size)
  end
end
```